liboptv/src/imgcoord.c

```diff
@@ -0,0 +1,89 @@
+/****************************************************************************
+
+Routine:	       	imgcoord.c
+
+Author/Copyright:      	Hans-Gerd Maas
+
+Address:	       	Institute of Geodesy and Photogrammetry
+                        ETH - Hoenggerberg
+		       	CH - 8093 Zurich
+
+Creation Date:	       	22.4.88
+
+Description:	       	computes x', y' from given Point and orientation
+	       		(see: Kraus)
+
+Routines contained:
+
+****************************************************************************/
+
+#include "imgcoord.h"
+
+
+/* flat_image_coord() calculates projection from coordinates in
+    world space to metric coordinates in image space without 
+    distortions
+    Arguments:
+    doubles X,Y,Z in real space
+    Calibration *cal parameters pointer
+    multimedia *mm parameters pointer
+    int i_cam - camera number (from 0 to cpar->num_cams)
+    multimedia look-up table array mmLUT pointer
+    Output:
+    double x,y in pixel coordinates in the image space
+ */
+ 
+void flat_image_coord (vec3d pos, Calibration *cal, mm_np *mm, double *x, double *y){
+
+  double deno;
+  Exterior Ex_t;
+  double X_t,Y_t,Z_t,cross_p[3],cross_c[3];
+  vec3d pos_t;
+
+  /* calculate tilted positions and copy them to X_t, Y_t and Z_t */
+  
+	trans_Cam_Point(cal->ext_par, *mm, cal->glass_par, pos, \
+          &Ex_t, pos_t, cross_p, cross_c);
+    
+    multimed_nlay (cal, mm, pos_t, &X_t,&Y_t);
```

Comment:
This is sending the wrong cal, because Ex_t is what ``multimed_nlay`` et al. need. Propose code like this:

```
Calibration cal_t; // instead of Ex_t
cal_t.mmlut = cal->mmlut;

trans_Cam_Point(cal->ext_par, *mm, cal->glass_par, pos, \
         &(cal_t.ext_par), pos_t, cross_p, cross_c);
multimed_nlay (&cal_t, mm, pos_t, &X_t,&Y_t);
```

It is a bug that I now spent about a day trying to find. So I think there is no point proceeding without also writing a *proper* test for at least flat_image_coord, not just a regression test that checks the result hasn't changed since the original submission.